Team417_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team417_CENTERSTAGE/mechanisms/ArmTeleOp.java

```diff
@@ -0,0 +1,249 @@
+                break;
+            }
+        }
+
+        return lastPosition;
+    }
+
+    public void armControl() {
+        //How much moving the right stick will effect arm speed outside and inside the robot.
+        double rStickSensitivity = 1, rStickSensitivityInsideRobot = rStickSensitivity / 3, rStickSensitivityOutsideRobot = rStickSensitivity;
+
+        double[] ArmPositions = new double[] {BaseOpMode.ARM_MOTOR_MIN_POSITION, BaseOpMode.ARM_MOTOR_MAX_POSITION / 2.0,
+                //array of arm positions the dpad can move to.
+                BaseOpMode.ARM_MOTOR_MAX_POSITION * (3.0/4.0), BaseOpMode.ARM_MOTOR_MAX_POSITION};
+
+        //The current distance the arm is from its init location
+        double armCurrentLocation = armMotor.getCurrentPosition();
+
+        //The amount the right stick has been moved
+        double rStickTilt = -gamepad2.right_stick_y;
+
+        //Defines a 0.1 deadzone for right stick input; allows a 25-unit error for arm position goals.
+        final double rStickDeadZone = 0.1, armPosEpsilon = 25;
+
+        //Locations the dumper should tilt so the arm does not hit the robot.
+        final double startUpwardTiltPos = 75, startUpwardResetPos = 1500, endUpwardResetPos = 2000, startDownwardTiltPos = 1500, endDownwardTiltPos = 250;
+
+        //The speed the arm moves while in move to position mode
+        final double armMoveToPositionVelocity = 0.5;
+
+        //If the arm is outside the dead zone switch to using stick control
+        if (gamepad2.left_stick_y < -rStickDeadZone || gamepad2.left_stick_y > rStickDeadZone) {
+            usingStick = true;
+
+        //If dpad up pressed, move arm to next position in array with armPosEpsilon tolerance.
+        } else if (gamepad2.dpad_up && !dpadUpPressed) {
+            armGoalLocation = findNextPosition(ArmPositions, armCurrentLocation, armPosEpsilon);
+            usingStick = false;
+
+        //If dpad down pressed, move arm to last position in array with armPosEpsilon tolerance.
+        } else if (gamepad2.dpad_down && !dpadDownPressed) {
+            armGoalLocation = findLastPosition(ArmPositions, armCurrentLocation, armPosEpsilon);
+            usingStick = false;
+        }
+
+        //If arm is inside robot decrease the speed so arm can't hit robot.
+        if (armMotor.getCurrentPosition() < 150.0)
+            rStickSensitivity = rStickSensitivityInsideRobot;
+        else
+            rStickSensitivity = rStickSensitivityOutsideRobot;
+
+        //If using stick control update the speed of arm based on the stick tilt
+        if (usingStick) {
+            armMotor.setPower(rStickTilt * rStickSensitivity);
+
+            //If stick is inside dead zone set motor power to zero
+            if (rStickTilt < rStickDeadZone && rStickTilt > -rStickDeadZone)
+                armMotor.setPower(0);
+        } else {
+            //if the arm is within 50 encoder ticks of it's goal, stop it from moving.
```

Comment:
```suggestion
            // if the arm is within armPosEpsilon encoder ticks of its goal, stop it from moving.
```
Probably don't want to directly say 50 in case you update the variable